### ISSUE TITLE
swapNFTsForETH vulnerability

### ISSUE THREAD

**Issue by moodlezoup** [sudoswap/lssvm#20]
This [line](https://github.com/0xmons/lssvm/blob/main/src/LSSVMPair.sol#L353) does not work as intended. 

The [`delete`](https://docs.soliditylang.org/en/v0.8.10/types.html?highlight=delete#delete) keyword in Solidity has no effect on mappings, including mappings nested within structs. 

`nftToETHTradeIdSet` is an `EnumerableSet.UintSet`, so the `delete` will have no effect on the [`_indexes`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/EnumerableSet.sol#L45) mapping. `EnumerableSet` uses the `_indexes` mapping for [`_contains`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/EnumerableSet.sol#L109-L111), so `nftToETHTradeIdSet.contains(nftIds[i])` would continue to return true for all IDs in `nftIds` in perpetuity. 

In other words, after someone sells their NFT(s) to an `LSSVMPair` via the router contract, they could drain the contract of ETH by repeatedly calling `swapNFTsForETH` using the same IDs.
